snapshot/restore/restore.go

```diff
@@ -92,12 +99,14 @@ func Entry(ctx context.Context, rep repo.Repository, output Output, rootEntry fs
 		}
 	}
 
-	// Control the depth of a restore. Default (options.MaxDepth = 0) is to restore to full depth.
+	// Control the depth of a restore. Default (options.RestoreDirEntryAtDepth = maxint) is to restore to full depth.
 	currentdepth := int32(0)
 
-	c.q.EnqueueFront(ctx, func() error {
-		return errors.Wrap(c.copyEntry(ctx, rootEntry, "", currentdepth, options.RestoreDirEntryAtDepth, func() error { return nil }), "error copying")
-	})
+	for _, ep := range roots {
```

Comment:
i think you need to capture ep for the closure:

```
ep := ep
```

Otherwise you'll be processing the last item in the slice.

Reply:
Oops. But probably moot given your other comment. 🙂